fundamentals/aot/diagnostics/Rdg7/Program.cs

```diff
@@ -0,0 +1,73 @@
+#define RDG11F //   RDG11 RDG11F
+#if NEVER
+#elif RDG11
+// <snippet_1>
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.TypeInfoResolverChain.Insert(0,
+                                 AppJsonSerializerContext.Default);
+});
+
+var app = builder.Build();
+
+app.MapPut("/v1/todos/{id}", ([AsParameters] TodoRequest request)
+    => Results.Ok(request.Todo));
+
+app.Run();
+
+public class TodoRequest
+{
+    public int Id { get; set; }
+    public Todo Todo { get; set; }
+    
+    private TodoRequest()
+    {
+    }
+}
+
+public record Todo(int Id, string Task);
+
+[JsonSerializable(typeof(Todo[]))]
+internal partial class AppJsonSerializerContext : JsonSerializerContext
```

Comment:
I based myself on other examples, but I think this isn't required?

Reply:
It was needed in the first sample and I just copied it. Maybe I can get rid of it in all my samples.